### PR TITLE
Fixed some Javadoc warnings ref issue #4995

### PR DESCRIPTION
**Pull request by joelmheim**
Feel free to use this if you like. Found a way to escape the warnings when using $var in comments.

### CODE REVIEW

File: framework/src/iteratees/src/main/scala/play/api/libs/iteratee/Enumeratee.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ object Enumeratee {
    * @param inner1 The first Iteratee to combine.
    * @param inner2 The second Iteratee to combine.
    * @param zipper Used to combine the results of each Iteratee.
-   * $paramEcSingle
+   * \$paramEcSingle
```
Comment (jroper) [inline framework/src/iteratees/src/main/scala/play/api/libs/iteratee/Enumeratee.scala]: I don't think this should be escaped, it's actually used to document the execution context param in one place, see here:

https://www.playframework.com/documentation/2.4.x/api/scala/index.html#play.api.libs.iteratee.Enumeratee$@zipWith[E,A,B,C](Iteratee[E,A],Iteratee[E,B])((A,B)⇒C)(ExecutionContext):Iteratee[E,C]

Comment (joelmheim) [inline framework/src/iteratees/src/main/scala/play/api/libs/iteratee/Enumeratee.scala]: Sorry for the late response to this. I will remove the incorrectly escaped dollar signs, but I'm having a little trouble checking in the code at the moment. Will update the PR as soon as possible.

**Comment by cchantep** [comment]
Hi, thanks for the contribution. Please squash the commits into a single one.

**Comment by joelmheim** [comment]
Done. Sorry 'bout that.

**Comment by richdougherty** [comment]
@joelmheim, the changes look good to me, thanks! There are some conflicts now, possibly caused by the merge of #5199.

**Comment by benmccann** [comment]
@joelmheim could you rebase so that we can merge?

File: framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSConfig.scala [hunk 2]
```diff
@@ -57,7 +57,7 @@ object CORSConfig {
       preflightMaxAge = 0.seconds)
 
   /**
-   * Build a [[CORSConfig]] from a [[play.api.Configuration]]
+   * Build a from a [[Configuration]]
```
Comment (benmccann) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSConfig.scala]: Looks like an accidental deletion here - "Build a from a"?

**Comment by benmccann** [comment]
Thanks @joelmheim !  Just one last comment.

P.S. leave a comment after you make the change so that we get notified to take another look. Thanks!

File: framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSConfig.scala (hunk 2 above)
Comment (marcospereira) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSConfig.scala]: Hey @joelmheim, could you please take a look at this? After that, I think this PR will be ok to merge.

Comment (mkurz) [inline framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSConfig.scala]: Ping @joelmheim

**Comment by joelmheim** [comment]
Updated the javadoc for CORSConfig as requested.

**Comment by mkurz** [comment]
@joelmheim Can you please squash your commits into a single one?

**Comment by joelmheim** [comment]
Done

**Pull request merged**